iotdb-core/ainode/ainode/core/manager/inference_manager.py

```diff
@@ -135,49 +141,167 @@ def infer(self, full_data, window_interval=None, window_step=None, **_):
-        for idx in range(self.DEFAULT_POOL_SIZE):
-            sundial_config = SundialConfig()
-            request_queue = mp.Queue()
-            request_pool = InferenceRequestPool(
-                pool_id=idx,
-                model_id=self.ACCELERATE_MODEL_ID,
-                config=sundial_config,
-                request_queue=request_queue,
+        system_res = self._evaluate_system_resources()
+        free_mem = system_res["free_mem"]
+
+        mem_usage = (
+            self.MODEL_MEM_USAGE_MAP[model_id] * self.INFERENCE_WEIGHT_OVERHEAD_RATIO
+        )
+        size = int((free_mem * self.INFERENCE_MEMORY_USAGE_RATIO) // mem_usage)
+        if size <= 0:
+            logger.error(
+                f"[Inference][Device-{self.DEFAULT_DEVICE}] Not enough memory to run model {model_id}. free={free_mem/1024**2:.2f} MB, need>={mem_usage/1024**2:.2f} MB"
+            )
+            return 0
+
+        logger.info(
+            f"[Inference][Device-{self.DEFAULT_DEVICE}] "
+            f"model={model_id}, mem_usage={mem_usage/1024**2:.2f} MB, "
+            f"pool_num={size}"
+        )
+        return size
+
+    def _first_pool_init(self, model_id: str):
+        if model_id == "sundial":
+            config = SundialConfig()
+        elif model_id == "timer_xl":
+            config = TimerConfig()
+        first_queue = mp.Queue()
+        ready_event = mp.Event()
+        first_pool = InferenceRequestPool(
+            pool_id=0,
+            model_id=model_id,
+            config=config,
+            request_queue=first_queue,
+            result_queue=self._result_queue,
+            ready_event=ready_event,
+        )
+        first_pool.start()
+        if not ready_event.wait(timeout=30):
+            logger.error(
+                f"[Inference][Device-{self.DEFAULT_DEVICE}][Pool-0] First pool failed to be ready in time"
+            )
+        else:
+            self._request_pool_map[model_id] = [(first_pool, first_queue)]
+            logger.info(
+                f"[Inference][Device-{self.DEFAULT_DEVICE}][Pool-0] Initialized inference request pool for model {model_id}"
+            )
+
+    def _expand_pools(self, model_id, start_idx, count):
+        for idx in range(count):
+            queue = mp.Queue()
+            if model_id == "sundial":
+                config = SundialConfig()
+            elif model_id == "timer_xl":
+                config = TimerConfig()
+            pool = InferenceRequestPool(
+                pool_id=start_idx + idx,
+                model_id=model_id,
+                config=config,
+                request_queue=queue,
                 result_queue=self._result_queue,
+                ready_event=mp.Event(),
             )
-            request_pool.start()
-            self._request_pool_map[self.ACCELERATE_MODEL_ID].append(
-                (request_pool, request_queue)
+            pool.start()
+            self._request_pool_map[model_id].append((pool, queue))
```

Comment:
The `ready_event` judgement should also be set here. Otherwise, the inference request might be routed to a unready pool